lib/brexit_checker/group.rb

```diff
@@ -0,0 +1,27 @@
+class BrexitChecker::Group
+  include ActiveModel::Validations
+
+  GROUPS_PATH = Rails.root.join("lib/brexit_checker/groups.yaml")
+
+  validates_inclusion_of :key, in: %w(visiting-eu
+                                      visiting-uk
+                                      visiting-ie
+                                      living-eu
+                                      living-ie
+                                      living-uk
+                                      working-uk
+                                      studying-eu
+                                      studying-uk)
+
+  attr_reader :key, :text
+
+  def initialize(attrs)
+    attrs.each { |key, value| instance_variable_set("@#{key}", value) }
+    validate!
+  end
+
+  def self.load_all
+    @load_all = nil if Rails.env.development?
```

Comment:
That's a really good question, I don't actually know...
I was just copy pasting this over from [criterion.rb:16](https://github.com/alphagov/finder-frontend/blob/858d7fa69a262235ab1ad138a39269e308863c43/lib/brexit_checker/criterion.rb#L16) on the basis that the two do something very similar:
- load data in from YAML,
- provide an interface to use it.

Had assumed there was a good reason for this?
Perhaps don't memoise for tests?

Reply:
actually no test is a separate environment... i wonder if there's another reason we don't want this in dev... @koetsier any idea why this would be in criterion?

Reply:
This ensures that in development mode it reloads the group file. So if Rails is running in development mode, changes in the groups.yaml file will be immediately visible without having to restart Rails.

Reply:
Cool, It makes sense here too? I'll leave it?

Reply:
Ok cool - would you mind popping a note about it in the commit message?

Reply:
Good plan, done [that here](https://github.com/alphagov/finder-frontend/pull/1776/commits/e3a759b83fae903ed784808702a1ff2e58fede9a)